pkg/proxy/ipvs/proxier.go

```diff
@@ -644,29 +638,31 @@ func (handle *LinuxKernelHandler) GetModules() ([]string, error) {
 		return nil, fmt.Errorf("failed to find loaded kernel modules: %v", err)
 	}
 
-	builtinModsFilePath := fmt.Sprintf("/lib/modules/%s/modules.builtin", kernelVersionStr)
-	b, err := ioutil.ReadFile(builtinModsFilePath)
+	bmods, nfconntrack, err = utilipvs.BuiltinIPVSModules(false, kernelVersionStr)
 	if err != nil {
-		klog.Warningf("Failed to read file %s with error %v. You can ignore this message when kube-proxy is running inside container without mounting /lib/modules", builtinModsFilePath, err)
+		return nil, fmt.Errorf("Failed to get kernel built-in modules")
 	}
 
+	mods = append(mods, bmods...)
 	for _, module := range ipvsModules {
-		if match, _ := regexp.Match(module+".ko", b); match {
-			bmods = append(bmods, module)
+		// Try to load the required IPVS kernel modules if not built in
```

Comment:
So in this part, there's something "annoying" to me:

If the module is built-in (in other part of the code, also) we still trigger an issue instead of just removing the bmods from the ipvsmodules or the ipvsmodules.

There's a more "elegant" part of the code that uses a (loadModules := sets.NewString()) so I'm wondering if this same strategy couldn't be used here, so:

1) We could use a set to effectively say "what modules are already loaded" 
2) If the Set does not contain the nfconntrack module, then we trigger the warning

Finally, we can return this set as a slice with the set.UnsortedSlice and the function signature won't change

@danwinship don't know if this makes sense, but I guess we could try this approach and change the mods to be a Set, and finally check on modprobe only modules that weren't effectively loaded as builtin modules.